### DISCUSSION TITLE
Incorporating User Testing to Our Current Workflow

### DISCUSSION THREAD

**Discussion by Brennan1994** [HydrologicEngineeringCenter/HEC-FDA#177] [General]
@codymccoy is close to finishing task1, and he and I were discussing the best way to get his stuff back into main. 

I suggested that when a branch is ready to be merged back into main, a pull request be made. When that pull request is made, the workflow (CI) will run. The owner of the branch will then have to work through any merge conflicts or failing tests. Once the tests all pass, the workflow will produce a zipped package of the application that can be grabbed by the Quality Control Tester and tested. if it is to be user tested, then the person responsible should be requested to give their review. Any changes due to their review can be made on the same branch for the same pull request. Once that is satisfied, they should give their approval, and assign Richard to give the final code review, and merge. 

I think this makes the most sense. I'm open to suggestion. The beauracracy for merging a branch becomes

- pull request
- satisfy automated testing
- assign user tester as reviewer
- satisfy user tests
- assign Project Lead as reviewer
- satisfy Project lead comments
- Project Lead Merges.

**Comment by rnugent3** [comment]
This is great! Thanks, @Brennan1994 and @codymccoy.
